tests/Doctrine/Tests/ORM/EntityManagerTest.php

```diff
@@ -288,4 +291,34 @@ public function testGetReferenceRetrievesReferencesWithGivenProxiesAsIdentifiers
         self::assertTrue($this->em->contains($simpleIdReferenceB));
         self::assertTrue($this->em->contains($nestedIdReference));
     }
+
+    public function testCallingFlushInsideFlush()
+    {
+        $listener = new class($this)
+        {
+            /** @var OrmTestCase */
+            private $testCase;
+
+            /** @var bool This flag is used to prevent infinity loop in case of regression */
+            private $alreadyInPreFlush = false;
+
+            public function __construct(OrmTestCase $testCase)
+            {
+                $this->testCase = $testCase;
+            }
+
+            public function preFlush(PreFlushEventArgs $eventArgs)
+            {
+                if ($this->alreadyInPreFlush) {
+                    $this->testCase->fail('Flush inside flush called. It should not happened. Fail test to prevent infinity loop');
+                }
+                $this->alreadyInPreFlush = true;
+                $eventArgs->getEntityManager()->flush();
+            }
+        };
+
+        $this->expectException(LogicException::class);
```

Comment:
As discussed in the thread, a custom exception is to be expected here

Reply:
done, used new one but can reuse any existing